Document GHC-59840 (#450)
haskellfoundation/error-message-index: message-index/messages/GHC-59840/index.md

---
title: GHC does not support GADTs or type families which witness equality of multiplicities
summary: GHC does not support GADTs or type families which witness equality of multiplicities
severity: error
introduced: 9.6.1
---

There are two features which GHC supports, and whose interaction is problematic and leads to this error:

- Programming constructs which build upon the type equality constraint `a ~ b`, such as GADTs and type families.

> - Linear types, which supports functions like `a %m -> b`. In this function type `m` stands for a *multiplicity* which can be either `One`(also written `1`) or `Many`, depending on how often the argument to the function is used in the body of the function.

Reviewer: ```suggestion
- Linear types, which supports functions like `a %m -> b`. In this function type `m` stands for a *multiplicity* which can be either `One`(also written `1`) or `Many`, which determines how often the argument to the function may be used in the body of the function.
```
